### DISCUSSION TITLE
Feeling a bit confused, is it possible to directly use the native web microphone recording interface?

### DISCUSSION THREAD

**Discussion by blackrabbit944** [ionic-team/capacitor#7685] [General]
When reading the official documentation,https://capacitorjs.com/docs/plugins/tutorial/designing-the-plugin-api ,it mentioned:

> Before building a plugin for a particular feature, we recommend checking out sites such as [What Web Can Do Today](https://whatwebcando.today/) to see if the functionality you are looking for is already available as a Web API.

Since I’ve already implemented the recording function using the native web interface and sent PCM data to the server, I’m a bit confused here. If the Web API can already be used directly, why would there be a need to provide an additional plugin?

However, in my actual test code, the navigator.mediaDevices object is undefined… So, I feel very confused. Is it really possible to use the web interface, or is a native implementation required?”

**Comment by jcesarmobile** [comment]
`navigator.mediaDevices` is not available on all iOS versions, I think it was added in some minor iOS 14 update, and needs permissions/usage descriptions to be added in the `AndroidManifest.xml`/`Info.plist` files.
